azure/packages/azure-client/src/test/AzureClient.spec.ts

```diff
@@ -45,239 +45,614 @@ describe("AzureClient", () => {
+            await new Promise<void>((resolve) => {
+                containerCopy.on("connected", () => {
+                    resolve();
+                });
+            });
+
+            assert.strictEqual(
+                typeof newContainerId,
+                "string",
+                "Attach did not return a string ID",
+            );
+            assert.strictEqual(
+                containerCopy.attachState,
+                AttachState.Attached,
+                "Container is not attached after attach is called",
+            );
+        });
+
+        /**
+         * Scenario: test if Azure Client can copy existing container at specific version.
+         *
+         * Expected behavior: an error should not be thrown nor should a rejected promise
+         * be returned.
+         */
+        it("can sucesfully copy document from a specific version", async () => {
+            const { container } = await client.createContainer(schema);
+            const containerId = await container.attach();
+
+            await new Promise<void>((resolve) => {
+                container.on("connected", () => {
+                    resolve();
+                });
+            });
+
+            const versions = await client.getContainerVersions(containerId);
+            assert.strictEqual(versions.length, 1, "Container should have exactly one version.");
+
+            const resources = client.copyContainer(containerId, schema, versions[0]);
+            await assert.doesNotReject(resources, () => true, "container could not be copied");
+
+            const { container: containerCopy } = await resources;
+
+            const newContainerId = await containerCopy.attach();
+            await new Promise<void>((resolve) => {
+                containerCopy.on("connected", () => {
+                    resolve();
+                });
+            });
+
+            assert.strictEqual(
+                typeof newContainerId,
+                "string",
+                "Attach did not return a string ID",
+            );
+            assert.strictEqual(
+                containerCopy.attachState,
+                AttachState.Attached,
+                "Container is not attached after attach is called",
+            );
+        });
+
+        /**
+         * Scenario: test if Azure Client properly handles DDS objects when
+         * copying existing container.
+         *
+         * Expected behavior: DDS values should match across original and copied
+         * container.
+         */
+        it("correctly copies DDS values when copying container", async () => {
+            const { container } = await client.createContainer(schema);
+
+            const initialObjectsCreate = container.initialObjects;
+            const map1Create = initialObjectsCreate.map1 as SharedMap;
+            map1Create.set("new-key", "new-value");
+            const valueCreate: string | undefined = await map1Create.get("new-key");
```

Comment:
Shouldn't need to await the get call

Reply:
good fined was a copy/past from earlier test case.